### DISCUSSION TITLE
Add to DataTable not from normal way (?)

### DISCUSSION THREAD

**Discussion by komidan** [Textualize/textual#3271] [Q&A] [answered]
Unsure on how to describe it in a title. Instead of doing the 'example' way of having tuples in an array like this: 
```py
ROWS = [
    ("lane", "swimmer", "country", "time"),
    (4, "Joseph Schooling", "Singapore", 50.39),
    (2, "Michael Phelps", "United States", 51.14),
    (5, "Chad le Clos", "South Africa", 51.14),
    (6, "László Cseh", "Hungary", 51.14),
    (3, "Li Zhuhao", "China", 51.26),
    (8, "Mehdy Metella", "France", 51.58),
    (7, "Tom Shields", "United States", 51.73),
    (1, "Aleksandr Sadovnikov", "Russia", 51.84),
    (10, "Darren Burns", "Scotland", 51.84),
]
```
and using:
```py
table.add_columns(*ROWS[0])
table.add_rows(ROWS[1:])
```
to populate the table, I've structured the data in a different way which would use for loops (preferably), but I can't find any method to add row information to a specified column, instead it seems like I need to use the `ROWS` style of table.

Is there a method like `table.add_rows(ROWS[1:], (column to add info to))` or something?

I've resorted to:
```py
[['testTable', {'col1': ['5', '12'], 'col2': ['testExample.1', 'textExample.2']}], ['moreData', {'ID': ['1', '2', '3'], 'name': ['cookiezi', 'mathi', 'mrekk'], 'number': ['727', '727', '1000']}]]
```
as my "storage". This is an example that I've been using, and the code I've tried to use to is:
```py
def on_tree_node_selected(self, event: Tree.NodeSelected) -> None:
    if not str(event.node.label) == sys.argv[1]:
        table = self.query_one(DataTable)
        table.clear(columns=all)

        currentTable = tableArr[(event.node.id - 1)]
        for column in currentTable[1]:
	    pass
	    for row in dict(currentTable[1]).get(column):
		    pass
```
They are being passed because other options didn't work. Figured I'd just get someone more experienced to help fill in the gaps.

EDIT:
The reason adding row items to specific column is what I ask for is because this happens:
![image](https://github.com/Textualize/textual/assets/91817886/6748b5fa-176d-4093-b69c-3899a2c8f58c)
when the code to add the items to the DataTable is:
```py
currentTable = tableArr[(event.node.id - 1)]
for column in currentTable[1]:
    table.add_column(column)
    for row in dict(currentTable[1]).get(column):
        table.add_row(row)
```

**Comment by davep** [comment]
If I'm reading you correctly, you're asking if there's a method of adding a *column* of data to an otherwise empty `DataTable`; is that correct? If so then, no, there's no direct method for doing that.

**Reply by komidan** [reply]
No, add *row* information to a specified column that was already created.

**Reply by davep** [reply]
Let me try and phrase my understanding a different way: given the columns have already been established with `add_column` or `add_columns`,  but so far no rows have been added, you are asking if there's a method for adding rows of data but for a specific column?

**Reply by komidan** [reply]
Yes

**Reply by davep** [reply]
Right. Then no, there's no direct support for that in DataTable.

**Comment by TomJGooding** [comment, marked as answer]
DataTable has no direct method of adding a _column_ of data, so how to populate a table when the data is column rather than row-orientated?

I've created a quick example based on the data from the docs, but let me know if there is a more efficient way!

```python
from itertools import zip_longest

from textual.app import App, ComposeResult
from textual.widgets import DataTable

DATA = {
    "lane": [4, 2, 5, 6, 3, 8, 7, 1, 10],
    "swimmer": [
        "Joseph Schooling",
        "Michael Phelps",
        "Chad le Clos",
        "László Cseh",
        "Li Zhuhao",
        "Mehdy Metella",
        "Tom Shields",
        "Aleksandr Sadovnikov",
        "Darren Burns",
    ],
    "country": [
        "Singapore",
        "United States",
        "South Africa",
        "Hungary",
        "China",
        "France",
        "United States",
        "Russia",
        "Scotland",
    ],
    "time": [50.39, 51.14, 51.14, 51.14, 51.26, 51.58, 51.73, 51.84, 51.84],
}


class TableApp(App):
    def compose(self) -> ComposeResult:
        yield DataTable()

    def on_mount(self) -> None:
        table = self.query_one(DataTable)

        columns = DATA.keys()
        rows = [
            [value for value in row]
            for row in zip_longest(  # type: ignore[call-overload]
                *[DATA[column] for column in columns],
                fillvalue=None,
            )
        ]

        table.add_columns(*columns)
        table.add_rows(rows)


app = TableApp()
if __name__ == "__main__":
    app.run()
```

**Reply by komidan** [reply]
Thank you very much for the example code, was able to implement it into my code and get it working.

Much Appreciated.
